http/src/request/channel/create_pin.rs

```diff
@@ -51,4 +53,13 @@ impl<'a> CreatePin<'a> {
     }
 }
 
+impl<'a> AuditLogReason for CreatePin<'a> {
+    fn reason(mut self, reason: impl Into<String>) -> Result<Self, AuditLogReasonError> {
+        let reason = AuditLogReasonError::validate(reason.into())?;
+        self.reason.replace(reason);
```

Comment:
We can at least abstract these 2 lines into a function, I'd really prefer to avoid macros if we don't need them

Reply:
I was not too sure what you meant here, I tried something, but it may not have been what you meant.